File: frontend/src/types/task.ts
// 任务相关类型

export type TaskStatus = 'waiting' | 'running' | 'completed' | 'failed'

export interface Task {
  id: string
  name: string
  status: TaskStatus
  progress: number
  created_at?: string
  completed_at?: string
  error?: string
  // LLM 任务特有
  operation?: string
  target_file?: string
}

export interface TaskQueue {
  queue: string[]
  running: string | null
  completed: string[]
}

export interface ExecutionLog {
  id: string
  task_id: string
  level: 'info' | 'warning' | 'error' | 'success'
  message: string
  timestamp: string
}

// 任务队列 API 类型
export interface TaskQueueItem {
  task_id: string
  status: string
  template: string
  created_at?: string
  completed_at?: string
  error?: string
}

export interface TaskQueueListResponse {
  tasks: TaskQueueItem[]
  total: number
  running: number
}

export interface TaskSubmitRequest {
  template_category: string
  template_type: string
  project_id: string
  target_file?: string | null
  variables?: Record<string, unknown>
}

File: backend/models/task.py
"""墨韵 - 任务模型"""

from datetime import datetime
from enum import Enum

from pydantic import BaseModel


class TaskStatus(str, Enum):
    """任务状态"""
    PENDING = "pending"
    RUNNING = "running"
    COMPLETED = "completed"
    FAILED = "failed"
    CANCELLED = "cancelled"


class TaskModel(BaseModel):
    """任务模型"""
    task_id: str
    template_category: str
    template_type: str
    variables: dict = {}
    status: TaskStatus = TaskStatus.PENDING
    target_file: str | None = None
    progress: float = 0.0
    error: str | None = None
    created_at: datetime
    started_at: datetime | None = None
    completed_at: datetime | None = None
